hex/Cargo.toml

```diff
@@ -0,0 +1,30 @@
+[package]
+name = "bitcoin-hex"
```

Comment:
Why the change? I find it really strange to name a general-purpose crate like this..

Reply:
Discussed above, I commented on the thread above to point you at it. I thought from that thread that we had consensus, will wait for more input and can rename if required.